VideoLocker/src/main/java/org/edx/mobile/module/db/impl/DbHelper.java

```diff
@@ -109,12 +116,60 @@ public void onUpgrade(SQLiteDatabase db, int oldVersion, int newVersion) {
                 createAssessmentTable(db);
             }
 
+            if (oldVersion < 6) {
+                Cursor cursor = db.query(false, DbStructure.Table.DOWNLOADS,
+                        new String[]{DbStructure.Column.ID, DbStructure.Column.USERNAME,
+                                DbStructure.Column.FILEPATH}, null, null, null, null, null, null);
+                if (cursor != null) {
+                    final File appExternalDir = mContext.getExternalFilesDir(null).getParentFile();
+                    try {
+                        while (cursor.moveToNext()) {
+                            final int id = cursor.getInt(0);
+                            final String username = cursor.getString(1);
+                            final String filePath = cursor.getString(2);
+                            final String encryptedUsername = Sha1Util.SHA1(username);
+                            final String newFilePath = filePath.replace(
+                                    appExternalDir.getAbsolutePath() + "/" + username,
+                                    appExternalDir.getAbsolutePath() + "/"
+                                            + AppConstants.Folders.VIDEOS + "/"
+                                            + encryptedUsername);
```

Comment:
Since both of these paths should constitute the whole string, how about just creating a new string instead of replacing?

```` java
if (!(appExternalDir.getAbsolutePath() + '/' + username)
        .equals(filePath)) {
    continue;
}
final String newFilePath = appExternalDir.getAbsolutePath() + "/"
        + AppConstants.Folders.VIDEOS + "/"
        + encryptedUsername;
````